### ISSUE TITLE
USER32.dll and KERNELBASE.dll errors when running nunit-console as different users

### ISSUE THREAD

**Issue by ralu-d** [nunit/nunit#695]
I need nunit-console to be started with several different users, once a week, automatically. In order to solve this, I have created a Scheduled Task that starts a Powershell script in which PsExec triggers the execution of nunit-console with different users.
However, the nunit-console process is started and then immediately stopped, resulting in the following errors in Event Viewer:
Faulting application name: nunit-console.exe, version: 2.6.2.12296, time stamp: 0x50861a21
Faulting module name: KERNELBASE.dll, version: 6.2.9200.17313, time stamp: 0x5507a832
Exception code: 0xc0000142
Fault offset: 0x00000000000e2100
Faulting process id: 0x708
Faulting application start time: 0x01d0a9aaad26e515
Faulting application path: C:\Program Files (x86)\NUnit 2.6.2\bin\nunit-console.exe
Faulting module path: KERNELBASE.dll
Report Id: ebcdc8de-159d-11e5-9448-00155d01e43e
Faulting package full name: 
Faulting package-relative application ID: 
and
Faulting application name: conhost.exe, version: 6.2.9200.16579, time stamp: 0x516363f3
Faulting module name: USER32.dll, version: 6.2.9200.17313, time stamp: 0x5507a832
Exception code: 0xc0000142
Fault offset: 0x00000000000e2100
Faulting process id: 0xad4
Faulting application start time: 0x01d0a9aaad26e515
Faulting application path: C:\Windows\system32\conhost.exe
Faulting module path: USER32.dll
Report Id: eb353189-159d-11e5-9448-00155d01e43e
Faulting package full name: 
Faulting package-relative application ID: 
When I start the script manually (so not from the scheduled task), it works fine. I've not been able to identify what is causing this and I need a solution. Could this be some bug in nunit?
Is there any other way to run nunit-console as different users unattended? Any suggestions and help are welcome!
Thanks in advance!

**Comment by CharliePoole** [comment]
Make sure you are not doing anything that would cause a security error, like trying to load the tests (or NUnit itself) from a share. Make sure you can run the tests manually _as_the_same_users_ you are trying to use with the scheduled task.

While we will keep this as an issue, I think you may be better served by asking about this in front of a broader audience. Issues are generally things we need to fix and typically are seen only by the NUnit developers. Most issues are commented on by three or four people at most. OTOH, the nunit-discuss list has over a thousand subscribers and by posting there you may get some help from another NUnit user who has encountered the same problem.

**Comment by ralu-d** [comment]
Running the tests manually as the same users works well. Both NUnit and the tests are accessed from local paths. The PsExec command is also configured to run on localhost.
Thank you, Charlie. I already posted it on StackOverflow as well, I'll head over to nunit-discuss and post it there as well.

**Comment by rprouse** [comment]
Moving to 3.0RC for confirmation, we may move out based on our investigation.

@ralu-d, can you provide a link to the StackOverflow question so we can follow that discussion too?

**Comment by rprouse** [comment]
I have tested this using NUnit 3 and PsExec works fine. I did have to specify the working directory on the command line, otherwise the test result file was trying to write to C:\Windows\System32 and got an access denied.

The command line I used to test this is,

```
psexec -u test -w C:\src\github\nunit\bin\debug -e .\nunit-console.exe --wait .\nunit.engine.tests.dll
```

I am closing this issue. @ralu-d, if you can get back to us with reproducible steps, then we can re-open this issue.

**Comment by rprouse** [comment]
@ralu-d, I noticed after I closed this issue that the problem was running PsExec through a scheduled task so I tested that also. I created a batch file to run the above command and scheduled it. It also ran fine.

It might be a problem with the user you are running the scheduled task as (not the user you are switching to) or it could be a bug in NUnit 2.x which we won't fix since it is working in 3.0.
